src/views/cursos/components/EditCourseForm.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react'
+import {
+  CButton,
+  CFormLabel,
+  CFormInput,
+  CSpinner,
+  CForm,
+  CRow,
+  CFormTextarea,
+  CFormSelect,
+} from '@coreui/react'
+import { useHistory } from 'react-router-dom'
+import Swal from 'sweetalert2'
+import { useForm } from 'react-hook-form'
+import { getCategoriesCourses, updateCourse } from 'src/services/courseQueryMethods'
+
+export const EditCourseForm = ({ courseData }) => {
+  const [categories, setCategories] = useState([])
+  const [loading, setLoading] = useState(false)
+  const history = useHistory()
+
+  useEffect(() => {
+    async function fetchData() {
+      const categories = await getCategoriesCourses()
+      setCategories(categories)
+    }
```

Comment:
Por que que crias esta função dentro do useEffect?
Se assim for, poderias fazer uma função anónima dentro do próprio useEffect.

Reply:
poderia dazer mesmo isso, mas a refencia que estava a ler mostru-me esta opção
mas vou já alterar..